draft-ietf-mls-protocol.md

```diff
@@ -5231,6 +5273,43 @@ groups by exhausting all available KeyPackages. To prevent such a denial-of-serv
 attack, the KeyPackage publication system SHOULD rate-limit KeyPackage
 requests, especially if not authenticated.
 
+## Delivery Service Compromise
+
+MLS is designed to protect the confidentiality and integrity of
+the group data even in the face of a compromised DS. However, a compromised
+DNS can still mount some attacks. While it cannot forge messages,
+it can selectively delay or remove them. This can in some cases be
+observed by detecting gaps in the per-sender generation counter,
+though it may not always be possible to distinguish attack from message
+loss. In addition, the DS can permanently block messages to and from
+a group member. This will not always be detectable by other members.
+If an application uses the DS to resolve conflicts between
+simultaneous Commits (see {{sequencing}}), it is also possible for the
+DS to influence which Commit is applied, even to the point of
+preventing a member from ever having its Commit applied.
+
+When put together, these abilities potentially allow a DS to collude
+with a compromised member to defeat PCS by suppressing the valid
+Update and Commit messages sent by the uncompromised endpoint for the
+same user. Aside from the generation, MLS leaves loss detection up the
```

Comment:
If I understand correctly, the "compromised member" and "the uncompromised endpoint" are more or less the same here, aren't they? It might make sense to spell the scenario out a bit more.

Also "generation" refers to the generation in the messages, right? Since it's a relatively generic word, it might make sense to be explicit about that context.

Reply:
Agree that this is a little ambiguous.  Suggest something like:

> When put together, these abilities potentially allow a DS to collude with an attacker who has compromised a member's state to defeat PCS by suppressing the valid Update and Commit messages from the member that would lock out the attacker and update the member's leaf to a new, uncompromised state.